### PR TITLE
Update `MetadataTests.testRulesOptions()` to check entire rule file, remove some special cases

### PR DESCRIPTION
**Pull request by calda**
This PR updates `MetadataTests.testRulesOptions()` to check the entire `RuleName.swift` source file, rather than only checking the `FormatRule`'s implementation closure.

#1798 moved all of `organizeDeclarations`'s helpers into `OrganizeDeclarations.swift`. There were two other exceptions for helpers that were only used by a single rule:
 - the `modifierOrder` helper was only used by the `modifierOrder` rule
 - the `indexWhereLineShouldWrapInLine` / `indexWhereLineShouldWrap` helpers were only used by the `wrap` rule

This change lets us remove these special cases from `MetadataTests.testRulesOptions()`.

What do you think @nicklockwood? This seems like a nice way to take advantage of the new rule-per-file organization structure.

I also made a few other changes to take advantage of this new structure:
 - Moved several other rule-specific helpers from `ParsingHelpers` / `FormattingHelpers`. These felt tightly-coupled to the rule itself, so seemed like a better fit in those rule files.
 - Moved several helper functions currently defined as local functions inside the rule implementation closure to instead be defined in an extension at the bottom of the file. This makes it easier to understand the code flow of the rule, and makes the input to the helpers more explicit.

### CODE REVIEW

File: Tests/MetadataTests.swift [hunk 1]
```diff
@@ -222,39 +230,6 @@ class MetadataTests: XCTestCase {
...
-                            Descriptors.organizeEnumThreshold,
-                            Descriptors.organizeExtensionThreshold,
-                            Descriptors.lineAfterMarks,
-                            Descriptors.organizationMode,
-                            Descriptors.alphabeticallySortedDeclarationPatterns,
-                            Descriptors.visibilityOrder,
-                            Descriptors.typeOrder,
-                            Descriptors.customVisibilityMarks,
-                            Descriptors.customTypeMarks,
-                            Descriptors.blankLineAfterSubgroups,
```
Comment (miguel-jimenez-0529) [inline Tests/MetadataTests.swift:256]: Nice!!

**Review by miguel-jimenez-0529 — APPROVED** [review]

**Comment by nicklockwood** [comment]
@calda it makes sense. The only problems I see with moving helpers into the rule files themselves are:

1) It makes them less discoverable. We already have cases where rules don't make use of a helper and reinvent the wheel (sometimes buggily) either because the rule predates the helper, or the author forgot (or didn't know) that the helper existed

2) If someone does find a legitimate use for a currently rule-specific helper, they now need to hoist it out into the helpers file, which is an added burden (though I suppose that was already true for helpers implemented as local functions)

2) if we make the helpers fileprivate then they are harder to test, but if we don't then inevitably someone will (accidentally or out of laziness) call the helper in one rule directly from another rule without hoisting it, which is structurally undesirable

TBC I don't think any of these are concrete reasons *not* to do it, but they bother me a little.

**Comment by calda** [comment]
All good points.

For #3, I could write a check in `MetadaataTests` that verifies the helpers defined in the rule files aren't used by other rules. If somebody tries to re-use one of these the test case could say to move the helper to `FormattingHelpers` or `ParsingHelpers`. An upside of this approach is that it lets us keep the helpers `internal` so they still appear in autocomplete, which is more discoverable than if they were fileprivate.

Alternatively we could just make the helpers in rule file be fileprivate (with a test case enforcing this). This is probably a more obvious solution since it actually uses access control, but makes the helpers less discoverable.

Thoughts?

**Comment by calda** [comment]
Will go ahead and merge this, and I can add whichever enforcement mechanism we decide on in a follow-up.

**Comment by nicklockwood** [comment]
> Alternatively we could just make the helpers in rule file be fileprivate (with a test case enforcing this). This is probably a more obvious solution since it actually uses access control, but makes the helpers less discoverable.

As I said in point  3, the only downside of this is that we can't then add unit tests for those helpers as @ testable doesn't apply to fileprivate methods. The only way to test them would be implicitly via testing the rule itself (which is probably fine in most cases, but it might mean losing some tests we already have in the test suite, which would be a shame)

**Comment by nicklockwood** [comment]
> For #3, I could write a check in `MetadaataTests` that verifies the helpers defined in the rule files aren't used by other rules. If somebody tries to re-use one of these the test case could say to move the helper to `FormattingHelpers` or `ParsingHelpers`.

This sounds like a good solution 👍

**Pull request merged**